### DISCUSSION TITLE
Navigation item visibility

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6856] [Help]
Hi all,

I've not understood how role permission and  menu items visibility works;

I created three roles, System Administrator, Tenant Administrator and Customer

Assigned to System Administrator role the capability on all table except Customer table (visible only to TenantAdmin Role) and PecMailbox table (visible only to Customer role) (see attached images)

On _CustomerPage.cs_ added 
`[PageAuthorize(typeof(CustomerRow))]`

The same thing on _PecMailboxPage.cs_
`[PageAuthorize(typeof(PecMailboxRow))]`

The user admin 

This is my AdministrationNavigation.cs file

```
using PecDownloaderAdmin.Administration;
using Serenity.Navigation;
using Administration = PecDownloaderAdmin.Administration.Pages;

[assembly: NavigationMenu(9000, "Administration", icon: "fa-wrench")]
[assembly: NavigationLink(9400, "Administration/Sys Configs", typeof(Administration.SysConfigPage), icon: "fa-gears")]
[assembly: NavigationLink(9500, "Administration/Customers", typeof(Administration.CustomerPage), icon: "fa-building")]
[assembly: NavigationLink(9600, "Administration/Users Management", typeof(Administration.UserController), icon: "fa-users")]
[assembly: NavigationLink(9700, "Administration/Pec Mailboxes", typeof(Administration.PecMailboxPage), icon: "fa-envelope")]
[assembly: NavigationLink(9800, "Administration/Errors", typeof(Administration.ErrorPage), icon: "fa-exclamation-triangle")]
[assembly: NavigationLink(9900, "Administration/Audit Logs", typeof(Administration.AuditLogPage), icon: "fa-eye")]
```

the user admin has only System Administrator Role, but he see both Pec Mailboxes and Customers menu items; 

Why? 
What i'm missing?

any help will be appreciated.
![Schermata 2023-07-26 alle 10 00 46](https://github.com/serenity-is/Serenity/assets/14870650/026227c0-8096-4799-b399-2e807e8c5954)
![Schermata 2023-07-26 alle 10 01 17](https://github.com/serenity-is/Serenity/assets/14870650/fa7bc1ad-406c-4f5c-90af-abc781c7b4b8)
![Schermata 2023-07-26 alle 10 01 53](https://github.com/serenity-is/Serenity/assets/14870650/485d674c-5909-4e77-9a10-2996d1a7abba)
![Schermata 2023-07-26 alle 10 10 41](https://github.com/serenity-is/Serenity/assets/14870650/f6366a2e-d01d-4988-95d4-7c764408d968)

Thank you,
Elena.

**Comment by elenaaralla** [comment]
Some update
If I create a new user with Role "System Administrator", he sees the right menu item, i.e. all items except Customer and PecMailboxes
So in this case it works as expected.

I do not understand why, for the existing _admin_ user, the assignment of _System Administrator_ role does not limit the menu items visibility.

Any idea?
Thank you,
Eena.

**Comment by osmanaslancan** [comment]
admin is not checked for permission. If username is admin it has all the permissions. You can edit this behavior in your PermissionService.cs.

https://github.com/serenity-is/serene/blob/66f5937e8c1c5cd3e7889d0be29106ee2b55592c/src/Serene.Web/Modules/Administration/User/Authentication/PermissionService.cs#L45C18-L45C18

**Reply by elenaaralla** [reply]
Hi @osmanaslancan ,

I'li take a look at document 

thank you.

**Comment by elenaaralla** [comment]
Ok,

I have to comment this rows

```
            if (username == "admin")
                return true;
```

Thank you.
